### PR TITLE
refactor: Move some common files out of rdsn directory

### PR DESCRIPTION
**Pull request by acelyc111**
https://github.com/apache/incubator-pegasus/issues/1053
- move rdsn docs to ./docs
- move common *.thrift to ./idl
- move thrift compilation scripts to ./scripts

### CODE REVIEW

**Review by foreverneverer — APPROVED** [review]

File: src/rdsn/src/runtime/security/security.thrift [hunk 1]
```diff
@@ -15,7 +15,7 @@
 // specific language governing permissions and limitations
 // under the License.
 
-include "../../dsn.thrift"
+include "../../../../../idl/dsn.thrift"
```
Comment (neverchanje) [inline src/rdsn/src/runtime/security/security.thrift:18]: dsn.thrift in rdsn/src/ looks fine to me, why did you move it to an outer directory?

**Review by neverchanje — APPROVED** [review]
rest lgtm

**Comment by neverchanje** [comment]
```
[ RUN      ] meta_bulk_load_http_test.start_compaction_test
got signal id: 11
Segmentation fault (core dumped)
```

Seems bulkload has some bugs.

File: src/rdsn/src/runtime/security/security.thrift (hunk 1 above)
Comment (acelyc111) [reply src/rdsn/src/runtime/security/security.thrift:18]: dsn.thrift contains some common thrift structures which will be used for not only rdsn, but also client libs. Its name 'dsn.thrift' is not accurate, maybe 'base.thrift' would be better, like it named in client lib directories(e.g. https://github.com/apache/incubator-pegasus/blob/master/java-client/idl/base.thrift).

**Comment by acelyc111** [comment]
> start_compaction_test

not related to this pr, we tracked it by issue, https://github.com/apache/incubator-pegasus/issues/1060

File: src/rdsn/src/runtime/security/security.thrift (hunk 1 above)
Comment (acelyc111) [reply src/rdsn/src/runtime/security/security.thrift:18]: Of course, we will refactor the code and only leave one unique base.thrift in Pegasus later.
